techlibs/xilinx/synth_xilinx.cc

```diff
@@ -251,16 +264,15 @@ struct SynthXilinxPass : public Pass
 
 		if (check_label(active, run_from, run_to, "fine"))
 		{
-			Pass::call(design, "opt -fast -full");
+			Pass::call(design, "opt -fast");
 			Pass::call(design, "memory_map");
 			Pass::call(design, "dffsr2dff");
 			Pass::call(design, "dff2dffe");
-			Pass::call(design, "opt -full");
```

Comment:
I guess `-full` is removed here in an attempt to make `pmux2shiftx` below work. But the correct solution would be to move `pmux2shiftx` up to the coarse-grain phase where it belongs.

Reply:
Yes, and the moving of `techmap -map +/techmap.v` down to just before `abc`.